api/genomics/docs/uploading_1kg_wgs_test_data.md

```diff
@@ -19,14 +19,23 @@ were copied into `gs://all-of-us-workbench-test-genomics/1kg_gvcfs/`
   - The first column represents sample_ids and will be incrementing by 100 in order to hit more partitions as part of this test
   - The second column represents sample_names and it will match the names in the reblocked vcf files
   - `cat 1kg_reblocked_gvcfs_list.txt | pcregrep -o1 'gvcf/(.+).haplo' | awk '{print NR*100","$0}' > 1kg_sample_map.txt`
+  - ImportGenomes wdl - https://github.com/broadinstitute/gatk/blob/75f0bd80a45ed46e171ba2125fcb6a6b825c1b74/scripts/variantstore/wdl/ImportGenomes.wdl
 
 ### Fixing sample_names
 - Right now, the sample_names in the `metadata` table match the sample_names in the VCFs
 - However, we need the sample_names to match participant_ids in AoU
 
 - ```bq query --max_rows=1000000 --format=csv --nouse_legacy_sql 'SELECT person_id FROM `all-of-us-workbench-test.synth_r_2019q4_9.cb_person`;' | sed -e 1d > ehr_sample_names.txt```
-- ```bq query --format=csv --nouse_legacy_sql 'SELECT sample_name FROM `all-of-us-workbench-test.wgs_demo.metadata`' | sed -e 1d > vcf_sample_names.txt```
+- ```bq query --max_rows=1000000 --format=csv --nouse_legacy_sql 'SELECT sample_name FROM `all-of-us-workbench-test.wgs_demo.metadata`' | sed -e 1d > vcf_sample_names.txt```
 - ```paste -d "," <(shuf -n $(cat vcf_sample_names.txt | wc -l) ehr_sample_names.txt) vcf_sample_names.txt > old_new_sample_names_map.csv```
 - ```cat old_new_sample_names_map.csv | awk -F',' '{print "UPDATE `all-of-us-workbench-test.wgs_demo_7.metadata` set sample_name=\""$1"\" where sample_name=\""$2"\";"}'```
 - Run update queries in BQ console or bq CLI
 
+### Extracting cohort
+- The first few runs of the extract failed until I tweaked some parameters
+- The scatter count was increased to 1000 to let the extraction finish in a reasonable time frame (~3 hours). 
+- The extraction time per shard varies a lot right now. Probably due to some more complex genomic intervals containing many more variants than less complex regions.
+- Some of the shards were also failing until I turned down the parameter for max local records in memory from 10 million to 5 million.
```

Comment:
Failing == OOM, right? I'm wondering whether we should considering having this parameter be a function of the number samples in the cohort being extracted for the service method. Other than that, I'm imagining the memory overhead of an average row should be uniform across cohorts and datasets.

Reply:
I couldn't pinpoint is as OOM from the logs but the problem disappeared after changing this parameter so I think it's likely. We should adjust the parameter automatically once we figure out what the right coefficients are. I think the interval length and location probably also matter.